apps/prairielearn/src/migrations/20241007203121_draft_questions.sql

```diff
@@ -0,0 +1,34 @@
+ALTER TABLE questions
+ADD COLUMN IF NOT EXISTS is_draft boolean NOT NULL DEFAULT (false);
+
+ALTER TABLE pl_courses
+ADD COLUMN IF NOT EXISTS draft_number integer NOT NULL DEFAULT (0);
+
+CREATE TABLE IF NOT EXISTS questions_draft_metadata (
+  id bigint NOT NULL PRIMARY KEY,
+  draft_created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  draft_created_by BIGINT REFERENCES users (user_id) ON DELETE SET NULL ON UPDATE CASCADE,
+  draft_updated_by BIGINT REFERENCES users (user_id) ON DELETE SET NULL ON UPDATE CASCADE
+);
+
+CREATE TYPE ai_prompt_type AS ENUM(
+  'initial_prompt',
+  'human_revision',
+  'autorevision'
+);
+
+CREATE TABLE IF NOT EXISTS ai_generation_prompts (
+  id bigserial PRIMARY KEY,
+  question_id BIGINT REFERENCES questions (id) ON DELETE SET NULL ON UPDATE CASCADE,
+  prompting_user BIGINT REFERENCES users (user_id) ON DELETE SET NULL ON UPDATE CASCADE,
```

Comment:
See all our other tables, we don't use `..._user` in isolation, it's always `..._user_id`.

Since this is accessible from pages where users can be impersonated, I think we also want an `authn_user_id` to be involved. Honestly, we can probably just use `user_id` and `authn_user_id` without the `prompting_` prefix; that's what we do elsewhere.


```suggestion
  prompting_user_id BIGINT REFERENCES users (user_id) ON DELETE SET NULL ON UPDATE CASCADE,
```

Reply:
changed in 562ccc7ca4238b061cc5fed79757979f8c0db35b.

Reply:
@mwest1066 do you think it's worth also tracking `authn_user_id` here too?